src/ServarrAPI/Controllers/UpdateController.cs

```diff
@@ -181,5 +190,28 @@ private string GetCacheControlHeader(DateTime expiry)
 
             return $"public,s-maxage={maxage},max-age=0";
         }
+
+        private Version GetMaxVersion(string currentVersion)
+        {
+            Version maxVersion = null;
+
+            if (!string.IsNullOrWhiteSpace(currentVersion))
+            {
+                var installedVersion = new Version(currentVersion);
+                var gates = _config.VersionGates.OrderBy(x => x.MaxVersion);
```

Comment:
I think this could be simplified to
```
        private Version GetMaxVersion(string currentVersion)
        {
            if (!string.IsNullOrWhiteSpace(currentVersion))
            {
                var installedVersion = new Version(currentVersion);
                return _config.VersionGates.OrderBy(x => x.MaxVersion).FirstOrDefault(x => installedVersion <= x.MaxVersion)?.MaxUpgradeVersion;
            }
            else
            {
                return null;
            }
        }
```
Though I guess it depends on your view of Linq as to whether or not that's simpler :)